main.tf

```diff
@@ -78,49 +89,64 @@ module "alb" {
   tags = local.tags
 }
 
-module "ecr" {
-  source               = "git@github.com:byu-oit/terraform-aws-ecr?ref=v1.0.0"
-  name                 = var.app_name
-  image_tag_mutability = "IMMUTABLE"
-  tags = local.tags
+// Make sure the fargate task has access to get the parameters from the container secrets
+data "aws_iam_policy_document" "secrets_access" {
+  count = local.has_secrets ? 1 : 0
+
+  version = "2012-10-17"
+  statement {
+    effect = "Allow"
+    actions = [
+      "ssm:GetParameters",
+      "ssm:GetParameter",
+      "ssm:GetParemetersByPath"
+    ]
+    resources = local.secrets_arns
+  }
 }
+resource "aws_iam_policy" "secrets_access" {
+  count = local.has_secrets ? 1 : 0
 
-module "ecr_initial_image" {
-  source             = "git@github.com:byu-oit/terraform-aws-ecr-image?ref=v1.0.0"
-  dockerfile_dir     = var.dockerfile_dir
-  ecr_repository_url = module.ecr.repository.repository_url
-  docker_image_tag   = formatdate("YYYY-MM-DD_hh-mm", timestamp())
+  name   = "${var.app_name}_secrets_access"
+  policy = data.aws_iam_policy_document.secrets_access[0].json
 }
 
 module "fargate" {
-  source              = "git@github.com:byu-oit/terraform-aws-fargate.git?ref=v1.1.0"
+//  source              = "../terraform-aws-fargate"
+  source              = "git@github.com:byu-oit/terraform-aws-fargate.git?ref=v1.2.2"
   app_name            = var.app_name
   vpc_id              = module.acs.vpc.id
   subnet_ids          = module.acs.private_subnet_ids
   load_balancer_sg_id = module.alb.alb_security_group.id
+  security_groups     = var.security_groups
   target_groups = [
     {
       arn  = module.alb.target_groups["blue"].arn
       port = module.alb.target_groups["blue"].port
     }
   ]
-  container_image         = module.ecr_initial_image.ecr_image_url
+  task_cpu                = var.task_cpu
+  task_memory             = var.task_memory
+  container_image         = var.container_image_url
   container_env_variables = var.container_env_variables
   container_secrets       = var.container_secrets
-  task_policies           = var.task_policies
-
+  task_policies           = concat(length(aws_iam_policy.secrets_access) > 0 ? [aws_iam_policy.secrets_access[0].arn] : [], var.task_policies)
+  task_execution_policies = length(aws_iam_policy.secrets_access) > 0 ? [aws_iam_policy.secrets_access[0].arn] : []
```

Comment:
can you use `local.has_secrets` in your conditional statements here?

Reply:
It was to solve a weird problem I got doing a destroy